src/plone/restapi/deserializer/dxcontent.py

```diff
@@ -78,7 +81,20 @@ def __call__(self, validate_all=False, data=None):  # noqa: ignore=C901
                             'message': e.doc(), 'field': name, 'error': e})
                     else:
                         field_data[name] = value
-                        if value != dm.get():
+                        dm_value = dm.get()
+
+                        # This is required in case that we can compare
+                        # offset-naive and offset aware objects. We convert all
+                        # offset-naive datetimes to UTC timezone first and then
+                        # compare them
+                        if isinstance(value, datetime) and \
+                           isinstance(dm_value, datetime):
+                            if value.tzinfo is None:
+                                value = value.replace(tzinfo=pytz.UTC)
+                            if dm_value.tzinfo is None:
+                                dm_value = dm_value.replace(tzinfo=pytz.UTC)
```

Comment:
good catch that the second case in my proposal is unnessesary, two datetime objects that have the same time in utc are equal even if the contain different timezone infos.